include/soci/soci-platform.h

```diff
@@ -74,23 +74,31 @@ namespace std {
 }
 #endif
 
-//define DLL import/export on WIN32
 #ifdef _WIN32
 # ifndef _WIN32_WINNT
 #   define _WIN32_WINNT 0x0502 //_WIN32_WINNT_WS03, VS2015 support: https://msdn.microsoft.com/de-de/library/6sehtctf.aspx
-# endif // _WIN32_WINNT
+# endif
+
 # ifdef SOCI_DLL
-#  ifdef SOCI_SOURCE
-#   define SOCI_DECL __declspec(dllexport)
-#  else
-#   define SOCI_DECL __declspec(dllimport)
-#  endif // SOCI_SOURCE
-# endif // SOCI_DLL
-#endif // _WIN32
-//
-// If SOCI_DECL isn't defined yet define it now
-#ifndef SOCI_DECL
-# define SOCI_DECL
+#  define SOCI_DECL_EXPORT __declspec(dllexport)
+#  define SOCI_DECL_IMPORT __declspec(dllimport)
+# endif
+
+#elif defined(unix) || defined(__unix__) || defined(__unix) && ENABLE_VISIBILITY_SUPPORT
```

Comment:
This ought to be
```suggestion
#elif (defined(unix) || defined(__unix__) || defined(__unix)) && ENABLE_VISIBILITY_SUPPORT
```
to work correctly, `&&` has higher precedence than `||`.

But I wonder why do we need to test for Unix here, shouldn't CMake test fail (or maybe not even run) under non-Unix anyhow? I.e. maybe it could be just
```
#elif defined(SOCI_ENABLE_VISIBILITY_SUPPORT)`
```
?

Reply:
Hmmm, yeah.
Why check for unix when there is `ENABLE_VISIBILITY_SUPPORT`?
The code can definitly be simplified to `#elif defined(SOCI_ENABLE_VISIBILITY_SUPPORT)`